### ISSUE TITLE
Implement Video / Audio Improvements for EDU

### ISSUE THREAD

**Issue by adonahue** [rifflearning/zenhub-platform-project#87]
* As a learner, I can't use Riff to do my group work in the course, unless there is stable audio and video for my entire team. My team will pick a different format to meet if Riff doesn't work. 

* As a user, if Riff throttles my bandwidth automatically, and my video degrades too much, I will be mad that I don't have any control over that part of my user experience. 

The goal of this story is to implement a limited data stream for users in a Riff call, to force calls to stay within our minimum technical requirements of a network speed of 3 mbps up/down. By limiting the send rate of ALL participants' send rate, we effectively limit the rate that all the other peers need to consume and process. By reducing the processing load, we expect that in some cases it will improve call stability; primarily by having less data to process, and possibly, by reducing the load on the device's CPU, by virtue of having to process less data.  

Because of the P2P connections in Riff, we don't want to unnecessarily degrade video and quality when there are not many users. For this reason, we are taking a scaled approach to limiting user data. Fewer users should be able to process larger data streams, and the video tiles are also larger, making limited data more noticeable in the picture quality.

** Requested Functionality ** 
In calls w. two or three people - do not limit. 
In calls w. four people - limit each user to 900 mbps
In calls with five or more people - limit all users to 333 mbps

The stream for screen sharing should never be limited b/c the quality of the visual is so important to being able to read it. 
When screen sharing is used, all participants should be limited to 333. 
When screen sharing stops, users should return to their stream rate according to the rules above. 
When a user joins (including rejoining), all participants should be re-limited.

**Comment by mlippert** [comment]
I see a lot of goals but not actually any work, I'm guessing this is because what to do is dependent on what is determined by those spikes? In which case shouldn't they be listed as dependencies, and is this story ready to be in a sprint?

**Comment by jaedoucette** [comment]
Task breakdown:

- Frontloading of tasks (https://app.zenhub.com/workspaces/riff-projects-5cb6029b4103d93ceadf001b/issues/rifflearning/riff-rtc/132)

- The 5-ish things here.

- Test planning

**Comment by adonahue** [comment]
For the message display, please use the same design pattern that we use on the log in page: a rounded rectangle that contains text, dismissable by the user. 
![image.png](https://images.zenhubusercontent.com/5cbf1c836caba0089f532d44/415574e7-86df-4142-82a7-21f21717503e)

**Comment by ebporter** [comment]
I don't think we have a mechanism to systematically test this; our anecdotal testing by using it every day seems to be enough to accept this story. We can always reopen it or open a new story if we need to continue to work on or test this functionality.
